### PR TITLE
fix(magic-contracts): Add missing docstrings in magic contracts

### PR DESCRIPTION
**Pull request by Ginowine**
# Description of change

Some of the magic contract reference docs are missing information about using the functions. This PR adds Solidity docstrings to the functions of the following contracts:

- [x]  ISC
- [x]  ISCAccounts
- [x]  ISCPrivileged
- [x]  ISCSandbox
- [x]  ISCTypes
- [x]  ISCUtil

## Links to any relevant issues

`fixes issue #3422 

## Type of change

- Documentation Fix

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [x] I have followed the [contribution guidelines](/CONTRIBUTING.md) for this project
- [x] I have performed a self-review of my own code
- [x] I have selected the `develop` branch as the target branch

### CODE REVIEW

**Review by lucas-tortora — APPROVED** [review]

**Review by salaheldinsoliman — CHANGES_REQUESTED** [review]
@Ginowine LGTM! except for this small comment:
NatSpec tags are all optional. That is, if `@notice` explains what the function does, there is no need to add a `@dev`. only add it when there are extra details
Can you please give it another small iteration, where if a `@notice` suffices just remove `@dev`
Thanks

**Comment by Ginowine** [comment]
> @Ginowine LGTM! except for this small comment: NatSpec tags are all optional. That is, if `@notice` explains what the function does, there is no need to add a `@dev`. only add it when there are extra details Can you please give it another small iteration, where if a `@notice` suffices just remove `@dev` Thanks

@salaheldinsoliman. Ok great, thanks. I'll go through it again.

**Comment by lucas-tortora** [comment]
could you rename the PR to follow the [guidelines on confluence](https://iotafoundation.atlassian.net/wiki/spaces/IF/pages/2202402957/PR+Review+Guidelines)?

File: packages/vm/core/evm/iscmagic/ISCAccounts.sol [hunk 1]
```diff
@@ -10,31 +10,76 @@ import "./ISCTypes.sol";
...
+     * @dev This function retrieves the balance of L2 native tokens for a given account.
+     * @param id The ID of the native token
+     * @param agentID The ID of the agent (account) whose balance is to be retrieved
+     * @return The L2 native tokens balance of the specified account
+     */
     function getL2BalanceNativeTokens(NativeTokenID memory id, ISCAgentID memory agentID) external view returns (uint256);
 
-    // Get the L2 NFTs owned by an account
+    /**
+     * @dev This function retrieves the number of NFTs owned by a given account.
```
Comment (Dr-Electron) [inline packages/vm/core/evm/iscmagic/ISCAccounts.sol]:
```suggestion
     * @dev This function retrieves the IDs of NFTs owned by a given account.
```
An array of IDs is returned

File: packages/vm/core/evm/iscmagic/ISCAccounts.sol [hunk 2]
```diff
@@ -10,31 +10,76 @@ import "./ISCTypes.sol";
...
+     * @param agentID The ID of the agent (account) whose NFTs are to be retrieved
+     * @return An array of NFTIDs representing the NFTs owned by the specified account
+     */
     function getL2NFTs(ISCAgentID memory agentID) external view returns (NFTID[] memory);
 
-    // Get the amount of L2 NFTs owned by an account
+    /**
+     * @dev This function retrieves the NFTs owned by a given account.
```
Comment (Dr-Electron) [inline packages/vm/core/evm/iscmagic/ISCAccounts.sol]:
```suggestion
     * @dev This function retrieves the number of NFTs owned by a given account.
```

File: packages/vm/core/evm/iscmagic/ISCSandbox.sol [hunk 3]
```diff
@@ -10,50 +10,84 @@ import "./ISCTypes.sol";
...
     ) external view returns (ISCAssets memory);
 
-    // Send an on-ledger request (or a regular transaction to any L1 address).
-    // The specified `assets` are transferred from the caller's
-    // L2 account to the `evm` core contract's account.
-    // The sent request will have the `evm` core contract as sender. It will
-    // include the transferred `assets`.
-    // The specified `allowance` must not be greater than `assets`.
+    /**
+     * @dev Sends the specified assets from the caller's L2 account to the EVM core contract's account and includes the specified metadata and options.
```
Comment (Dr-Electron) [inline packages/vm/core/evm/iscmagic/ISCSandbox.sol]: I would rephrase this.
```suggestion
     * @dev Sends the specified assets from the caller's L2 account to a L1 address and includes the specified metadata and options. This can also be used to create on-ledger requests to the chain itself.
```

File: packages/vm/core/evm/iscmagic/ISCSandbox.sol [hunk 4]
```diff
@@ -62,55 +96,109 @@ interface ISCSandbox {
         ISCSendOptions memory sendOptions
     ) external payable;
 
-    // Call the entry point of an ISC contract on the same chain.
+    /**
+     * @dev Calls the specified entry point of the ISC contract with the given parameters and allowance.
+     * @param contractHname The hash name of the contract.
```
Comment (Dr-Electron) [inline packages/vm/core/evm/iscmagic/ISCSandbox.sol]: I would just use hname, that is what we use in the docs
```suggestion
     * @param contractHname The hname of the contract.
```

File: packages/vm/core/evm/iscmagic/ISCSandbox.sol [hunk 5]
```diff
@@ -62,55 +96,109 @@ interface ISCSandbox {
...
         ISCHname entryPoint,
         ISCDict memory params,
         ISCAssets memory allowance
     ) external returns (ISCDict memory);
 
-    // Call a view entry point of an ISC contract on the same chain.
-    // The called entry point will have the `evm` core contract as caller.
+    /**
+     * @dev Calls the specified view entry point of the ISC contract with the given parameters.
+     * @param contractHname The hash name of the contract.
```
Comment (Dr-Electron) [inline packages/vm/core/evm/iscmagic/ISCSandbox.sol]: Same here
```suggestion
     * @param contractHname The hname of the contract.
```

File: packages/vm/core/evm/iscmagic/ISCTypes.sol [hunk 6]
```diff
@@ -141,12 +141,23 @@ struct ISCTokenProperties {
 }
 
 library ISCTypes {
+    /**
+     * @dev Get the type of an L1 address.
+     * @param addr The L1 address.
+     * @return The type of the L1 address.
```
Comment (Dr-Electron) [inline packages/vm/core/evm/iscmagic/ISCTypes.sol:147]: I guess we could add a note here about address types mentioned in this section? Or just ad this link
https://wiki.iota.org/tips/tips/TIP-0018/#address-unlock-condition

**Review by Dr-Electron — CHANGES_REQUESTED** [review]

File: packages/vm/core/evm/iscmagic/ISCTypes.sol (hunk 6 above)
Comment (Ginowine) [reply packages/vm/core/evm/iscmagic/ISCTypes.sol:147]: Ok, thanks; I added a note and the link.

**Review by jorgemmsilva — APPROVED** [review]

**Review by Dr-Electron — APPROVED** [review]

**Review by salaheldinsoliman — APPROVED** [review]
